build/Build.Steps.Windows.cs

```diff
@@ -13,8 +15,8 @@ partial class Build
 {
     Target CompileNativeSrcWindows => _ => _
         .Unlisted()
+        .DependsOn(GenerateNetFxAssemblyRedirectionSource)
```

Comment:
Why this change? I do not remember the exact details right now, but there were some reasons that we are not using the `DependsOn` in "step targets". Reference: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/889

Reply:
This will decouple `GenerateNetFxAssemblyRedirectionSource` from `CompileNativeSrcWindows` imo?
`GenerateNetFxAssemblyRedirectionSource` is only connected to `CompileNativeSrcWindows`...

Reply:
`BuildTracer` is the "workflow" target which has the `DependsOn`. I propose to follow the current pattern otherwise we will have a mess.

Reply:
Which is exactly my point. I'm running these sub targets quite a lot. If you do clean and run `CompileNativeSrc` it won't compile due missing source file. Which means `CompileNativeSrc` does not do what it promises.

Reply:
In order to run `CompileNativeSrc` and get the correct bits under `tracer-home` one has to `BuildTracer` before (likely after a clean). My workflow is `BuildTracer` once and then build the components as needed, in this case, it will be nice to avoid the new target when doing `CompileNativeSrc`.

Reply:
I'll revert this part, not tightly coupled to this PR.

But in general I still don't agree. `CompileNativeSrcWindows` should do exactly what it promises, otherwise it's something else. If you need to optimize `CompileNativeSrcWindows` you can always do `nuke CompileNativeSrcWindows -skip GenerateNetFxAssemblyRedirectionSource` which then skips the parts and does personalized run.

Reply:
`nuke CompileNativeSrcWindows -skip` would even work

Reply:
I added https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/889#issuecomment-1406277116